lib/rules/operator-assignment.js

```diff
@@ -148,12 +109,12 @@ module.exports = {
             const operator = expr.operator;
 
             if (isCommutativeOperatorWithShorthand(operator) || isNonCommutativeOperatorWithShorthand(operator)) {
-                if (same(left, expr.left)) {
+                if (astUtils.isSameReference(left, expr.left, true)) {
                     context.report({
                         node,
                         messageId: "replaced",
                         fix(fixer) {
-                            if (canBeFixed(left)) {
+                            if (canBeFixed(left) && canBeFixed(expr.left)) {
```

Comment:
Is there a situation where `canBeFixed(left)` returns `true`, but `canBeFixed(expr.left)` returns `false`?

Reply:
I'm sorry, I cannot remember what I thought at that time... 😢 

Maybe, `a.b = a?.b + c` is reported and not fixed in the first implementation, but then I changed it and forgot to remove this line. I will change the rule to report `a.b = a?.b + c` later...

Reply:
I'm not sure if `operator-assignment` should report `a.b = a?.b + c`, since it looks more like a possible error than a stylistic issue, because this will throw if `a` is nullish.

And, it will be reported by `no-unsafe-optional-chaining` (maybe in this case even with `disallowArithmeticOperators: false`).

Reply:
As like `no-useless-escape` rule, it's hard to determine the writer's intention from `a.b = a?.b + c`; The `a` may be nullable (then throw) or the `?.` may be useless (then equal to `a.b`). Therefore, I think that both `operator-assignment` and `no-unsafe-optional-chaining` should report that.

Reply:
Agreed!